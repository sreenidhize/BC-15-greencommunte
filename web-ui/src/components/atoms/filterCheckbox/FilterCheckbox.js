import React from "react";
import { Checkbox, withStyles } from "@material-ui/core";

const FilterCheckbox = withStyles((theme) => ({
  root: {
    color: theme.palette.primary.main,
    "&$checked": {
      color: theme.palette.primary.main,
    },
  },
  checked: {},
}))((props) => (
  <Checkbox data-testid="filter-checkbox" color="default" {...props} />
));

export default FilterCheckbox;
