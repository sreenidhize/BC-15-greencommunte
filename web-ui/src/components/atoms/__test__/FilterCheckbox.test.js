import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import FilterCheckbox from "../filterCheckbox/FilterCheckbox";

describe("Filter checkbox", () => {
    it("Matches the snapshot", () => {
        const { container } = render(
            <FilterCheckbox />
        );
        expect(container).toMatchSnapshot();
    });

    test("test filter checkbox render", () => {
        const { getByTestId } = render(<FilterCheckbox />);
        expect(getByTestId("filter-checkbox")).toBeInTheDocument();
    });
});