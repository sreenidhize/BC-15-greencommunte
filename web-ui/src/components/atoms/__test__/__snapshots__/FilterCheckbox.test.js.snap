// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter checkbox Matches the snapshot 1`] = `
<div>
  <span
    aria-disabled="false"
    class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-3 MuiCheckbox-root Component-root-1"
    data-testid="filter-checkbox"
  >
    <span
      class="MuiIconButton-label"
    >
      <input
        class="PrivateSwitchBase-input-6"
        data-indeterminate="false"
        type="checkbox"
        value=""
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
        />
      </svg>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </span>
</div>
`;
